engines/router/missionctl/handlers/batch_http_handler.go

```diff
@@ -0,0 +1,124 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/gojek/turing/engines/router/missionctl"
+	"github.com/gojek/turing/engines/router/missionctl/errors"
+	"github.com/gojek/turing/engines/router/missionctl/log"
+	"github.com/gojek/turing/engines/router/missionctl/turingctx"
+)
+
+type batchHTTPHandler struct {
+	httpHandler
+}
+
+type batchResponse struct {
+	StatusCode int         `json:"code"`
+	ErrorMsg   string      `json:"error,omitempty"`
+	Data       interface{} `json:"data,omitempty"`
+}
+
+type batchResult struct {
+	BatchResult []batchResponse `json:"batch_result"`
+}
+
+// NewBatchHTTPHandler creates an instance of the Mission Control's prediction request handler
+func NewBatchHTTPHandler(mc missionctl.MissionControl) http.Handler {
+	return &batchHTTPHandler{httpHandler{mc}}
+}
+
+func (h *batchHTTPHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	var httpErr *errors.HTTPError
+	h.measureRequestDuration(httpErr)
+
+	// Create context from the request context
+	ctx := turingctx.NewTuringContext(req.Context())
+	// Create context logger
+	ctxLogger := log.WithContext(ctx)
+	defer func() {
+		_ = ctxLogger.Sync()
+	}()
+
+	// Get the unique turing request id from the context
+	turingReqID, err := turingctx.GetRequestID(ctx)
+	if err != nil {
+		ctxLogger.Errorf("Could not retrieve Turing Request ID from context: %v",
+			err.Error())
+	}
+	ctxLogger.Debugf("Received batch request for %v", turingReqID)
+
+	ctx = h.enableTracingSpan(ctx, req)
+
+	// Read the request body
+	requestBody, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		h.error(ctx, rw, errors.NewHTTPError(err))
+		return
+	}
+
+	//Valid request
+	if valid := json.Valid(requestBody); !valid {
+		h.error(ctx, rw, errors.NewHTTPError(errors.Newf(errors.BadInput,
+			`Invalid json request format`)))
+		return
+	}
+
+	//Split into batches
+	var batchRequests map[string][]interface{}
+	err = json.Unmarshal(requestBody, &batchRequests)
+	if err != nil {
+		h.error(ctx, rw, errors.NewHTTPError(errors.Newf(errors.BadInput,
+			`Invalid json request`)))
+		return
+	}
+
+	//Validate request
+	if _, ok := batchRequests["batch_request"]; !ok {
+		h.error(ctx, rw, errors.NewHTTPError(errors.Newf(errors.BadInput,
+			`batch_request" not found in request`)))
+		return
+	}
+
+	//Handle request asynchronously
+	var batchResponses = make([]batchResponse, len(batchRequests["batch_request"]))
+	var waitGroup sync.WaitGroup
+	for index, value := range batchRequests["batch_request"] {
+		waitGroup.Add(1)
+
+		go func(index int, jsonRequestBody interface{}) {
+			defer waitGroup.Done()
+			batchRequestBody, _ := json.Marshal(jsonRequestBody)
```

Comment:
That's very inefficient to unmarshal data first to marshal it again later. Think about the possible batch size in a magnitude of a thousand(s) predict requests. 

Maybe you can use `[]json.RawMessage` (instead of `[]interface{}`) to split the batch into individual requests?